### PR TITLE
Edit meeting

### PR DESCRIPTION
**Pull request by timothysashimi**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    }
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Meeting> lastShownList = model.getFilteredMeetingList();
+        /*
+        lastShownList.add(new Meeting("a", LocalDateTime.now(), selectedClient));
+        lastShownList.add(new Meeting("b", LocalDateTime.now(), selectedClient));
+        lastShownList.add(new Meeting("c", LocalDateTime.now(), selectedClient));
+        */
```
Comment (itstrueitstrueitsrealitsreal) [inline src/main/java/seedu/address/logic/commands/EditMeetingCommand.java]: Is there a reason why you left the commented out code in?

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+        Meeting editedMeeting = createEditedMeeting(meetingToEdit, editMeetingDescriptor);
+
+        if (!meetingToEdit.isSameMeeting(editedMeeting) && lastShownList.contains(editedMeeting)) {
+            throw new CommandException(MESSAGE_DUPLICATE_MEETING);
+        }
+        lastShownList.set(index.getZeroBased(), editedMeeting);
+        client.setMeetings(lastShownList);
+        //model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS, Messages.formatMeeting(editedMeeting)));
+    }
```
Comment (itstrueitstrueitsrealitsreal) [inline src/main/java/seedu/address/logic/commands/EditMeetingCommand.java]: Perhaps commented out code can be removed if not in use.

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java (hunk 1 above)
Comment (timothysashimi) [reply src/main/java/seedu/address/logic/commands/EditMeetingCommand.java]: yea i was trying to test the execution before the adding meeting command was added, i can remove it

File: src/main/java/seedu/address/logic/parser/EditMeetingCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                        PREFIX_NAME, PREFIX_DATETIME);
+
+        Map<Prefix, List<String>> map = argMultimap.getMap();
+        /*
+        for (Prefix name: map.keySet()) {
+            String key = name.toString();
+            String value = map.get(name).toString();
+            System.out.println(key + " " + value);
+        }
+        */
```
Comment (itstrueitstrueitsrealitsreal) [inline src/main/java/seedu/address/logic/parser/EditMeetingCommandParser.java:46]: remember to remove the commented out code!

File: src/main/java/seedu/address/logic/parser/EditMeetingCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_DATETIME, PREFIX_CLIENT);
+
+        EditMeetingCommand.EditMeetingDescriptor editMeetingDescriptor = new EditMeetingCommand.EditMeetingDescriptor();
+
+        if (argMultimap.getValue(PREFIX_CLIENT_INDEX).isPresent()) {
+            clientIndex = ParserUtil.parseClientIndex(argMultimap.getValue(PREFIX_CLIENT_INDEX).get());
+        }
+        if (argMultimap.getValue(PREFIX_MEETING_INDEX).isPresent()) {
+            meetingIndex = ParserUtil.parseMeetingIndex(argMultimap.getValue(PREFIX_MEETING_INDEX).get());
+            //System.out.println(meetingIndex); correct
```
Comment (itstrueitstrueitsrealitsreal) [inline src/main/java/seedu/address/logic/parser/EditMeetingCommandParser.java:74]: You could remove the logging statement here if it's not being used.

File: src/test/java/seedu/address/logic/commands/EditMeetingCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    /*
+    @Test
+    public void execute_duplicateMeetingUnfilteredList_failure() {
+        Meeting firstMeeting = model.getFilteredMeetingList().get(INDEX_FIRST_MEETING.getZeroBased());
+        EditMeetingCommand.EditMeetingDescriptor descriptor = new EditMeetingDescriptorBuilder(firstMeeting).build();
+        EditMeetingCommand editMeetingCommand = new EditMeetingCommand(INDEX_SECOND_PERSON,
+                descriptor, INDEX_SECOND_MEETING);
+
+        assertCommandFailure(editMeetingCommand, model, EditMeetingCommand.MESSAGE_DUPLICATE_MEETING);
+    }
```
Comment (itstrueitstrueitsrealitsreal) [inline src/test/java/seedu/address/logic/commands/EditMeetingCommandTest.java:115]: You could remove the commented out code here if possible too.

File: src/test/java/seedu/address/logic/commands/EditMeetingCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+        // different index -> returns false
+        assertFalse(standardCommand.equals(new EditMeetingCommand(INDEX_SECOND_PERSON,
+                new EditMeetingCommand.EditMeetingDescriptor(), INDEX_FIRST_MEETING)));
+
+        // different descriptor -> returns false
+        assertFalse(standardCommand.equals(new EditMeetingCommand(INDEX_FIRST_PERSON,
+                new EditMeetingCommand.EditMeetingDescriptor(), INDEX_SECOND_MEETING)));
+    }
+    */
```
Comment (itstrueitstrueitsrealitsreal) [inline src/test/java/seedu/address/logic/commands/EditMeetingCommandTest.java:183]: As mentioned above

File: src/test/java/seedu/address/logic/parser/EditMeetingCommandParserTest.java [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+    public void parse_missingParts_failure() {
+        // no index specified
+        assertParseFailure(parser, VALID_NAME_AMY, MESSAGE_INVALID_FORMAT);
+
+        // no field specified
+        assertParseFailure(parser, "1", EditMeetingCommand.MEETING_NOT_EDITED);
+
+        // no index and no field specified
+        assertParseFailure(parser, "", MESSAGE_INVALID_FORMAT);
+    }
```
Comment (itstrueitstrueitsrealitsreal) [inline src/test/java/seedu/address/logic/parser/EditMeetingCommandParserTest.java:31]: Remember to remove unused code

File: src/test/java/seedu/address/logic/parser/EditMeetingCommandParserTest.java [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+        // multiple invalid values
+        userInput = targetIndex.getOneBased() + INVALID_PHONE_DESC + INVALID_ADDRESS_DESC + INVALID_EMAIL_DESC
+                + INVALID_PHONE_DESC + INVALID_ADDRESS_DESC + INVALID_EMAIL_DESC;
+
+        assertParseFailure(parser, userInput,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS));
+    }
+    */
+
+}
```
Comment (itstrueitstrueitsrealitsreal) [inline src/test/java/seedu/address/logic/parser/EditMeetingCommandParserTest.java:177]: As mentioned above

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+    public static final String COMMAND_WORD = "editMeeting";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the meeting identified "
+            + "by the index number used in the displayed meeting list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_CLIENT_INDEX + "CLIENT INDEX] "
+            + "[" + PREFIX_MEETING_INDEX + "MEETING INDEX] "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_DATETIME + "DATETIME] \n"
```
Comment (itstrueitstrueitsrealitsreal) [inline src/main/java/seedu/address/logic/commands/EditMeetingCommand.java:39]: Perhaps the square brackets here could be removed to follow the format of the message usage of the other commands?

**Review by itstrueitstrueitsrealitsreal — COMMENTED** [review]
I think that maybe you could try to make the tests work instead of commenting out the failing test cases.

Other than the issue of commented out code and the standardization of formatting, LGTM!

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java (hunk 1 above)
Comment (itstrueitstrueitsrealitsreal) [reply src/main/java/seedu/address/logic/commands/EditMeetingCommand.java]: Alright, can just clean it up when you're free!

**Pull request closed without merging**